plugins/woocommerce/src/Internal/Fulfillments/Fulfillment.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * WooCommerce order fulfillments.
+ *
+ * The WooCommerce order fulfillments class gets contains fulfillment related properties and methods.
+ *
+ * @package WooCommerce\Classes
+ * @version 9.9.0
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Fulfillments;
+
+use Automattic\WooCommerce\Internal\DataStores\Fulfillments\FulfillmentsDataStore;
+
+defined( 'ABSPATH' ) || exit;
+
+
+/**
+ * WC Order Fulfillment Class
+ *
+ * @since 9.9.0
+ */
+class Fulfillment extends \WC_Data {
+	/**
+	 * Fulfillment ID.
+	 *
+	 * @var int
+	 */
+	protected int $fulfillment_id = 0;
```

Comment:
If we're extending `\WC_Data` all of the attributes need to be stored in `$data` instead of object properties. It is what allows tracking of changes while a WC_Data object is in a changed state prior to saving back to the DB.  The blended approach where some of the properties are duplicated, like the fulfillment_id and items, it will likely cause bugs down the line if we don't have confidence in which is the canonical source for the value.

Reply:
Yes, I realized it while building the REST API, as the current state of the data store doesn't allow me to query the DB with entity type and id. I will add a new interface that will contain the REST method <-> database bridging functions and extend the data store with their implementations.

Reply:
Addressed in https://github.com/woocommerce/woocommerce/pull/57429/commits/f07fcfde8335623fd46fd7adfdb1628421eaccf7.